ext/node/ops/v8.rs

```diff
@@ -30,3 +39,343 @@ pub fn op_v8_get_heap_statistics(
   buffer[12] = stats.used_global_handles_size() as f64;
   buffer[13] = stats.external_memory() as f64;
 }
+
+pub struct Serializer<'a> {
+  inner: v8::ValueSerializer<'a>,
+}
+
+pub struct SerializerDelegate {
+  obj: v8::Global<v8::Object>,
+}
+
+impl<'a> v8::cppgc::GarbageCollected for Serializer<'a> {
+  fn trace(&self, _visitor: &v8::cppgc::Visitor) {}
+}
+
+impl SerializerDelegate {
+  fn obj<'s>(
+    &self,
+    scope: &mut v8::HandleScope<'s>,
+  ) -> v8::Local<'s, v8::Object> {
+    v8::Local::new(scope, &self.obj)
+  }
+}
+
+impl v8::ValueSerializerImpl for SerializerDelegate {
+  fn get_shared_array_buffer_id<'s>(
+    &self,
+    _scope: &mut v8::HandleScope<'s>,
+    _shared_array_buffer: v8::Local<'s, v8::SharedArrayBuffer>,
+  ) -> Option<u32> {
+    None
+  }
+  fn has_custom_host_object(&self, _isolate: &mut v8::Isolate) -> bool {
+    false
+  }
+  fn throw_data_clone_error<'s>(
+    &self,
+    scope: &mut v8::HandleScope<'s>,
+    message: v8::Local<'s, v8::String>,
+  ) {
+    let obj = self.obj(scope);
+    let key = FastString::from_static("_getDataCloneError")
+      .v8_string(scope)
+      .into();
+    if let Some(v) = obj.get(scope, key) {
+      if let Ok(fun) = v.try_cast::<v8::Function>() {
```

Comment:
Shouldn't this be an error/panic if it's not a function?

Reply:
Ah yeah, it should panic (in node it's an assert)